packages/opentelemetry-metrics/src/export/types.ts

```diff
@@ -102,16 +113,68 @@ export interface MetricExporter {
 /**
  * Base interface for aggregators. Aggregators are responsible for holding
  * aggregated values and taking a snapshot of these values upon export.
+ *
+ * Use {@link Aggregator} instead of this BaseAggregator.
  */
-export interface Aggregator {
+interface BaseAggregator {
+  /** The kind of the aggregator. */
+  kind: AggregatorKind;
+
   /** Updates the current with the new value. */
   update(value: number): void;
+}
+
+/** SumAggregatorType aggregate values into a {@link Sum} point type. */
+export interface SumAggregatorType extends BaseAggregator {
+  kind: AggregatorKind.SUM;
+
+  /** Returns snapshot of the current point (value with timestamp). */
+  toPoint(): Point<Sum>;
+}
+
+/**
+ * LastValueAggregatorType aggregate values into a {@link LastValue} point
+ * type.
+ */
+export interface LastValueAggregatorType extends BaseAggregator {
+  kind: AggregatorKind.LAST_VALUE;
+
+  /** Returns snapshot of the current point (value with timestamp). */
+  toPoint(): Point<LastValue>;
+}
+
+/**
+ * DistributionAggregatorType aggregate values into a {@link Distribution}
+ * point type.
+ */
+export interface DistributionAggregatorType extends BaseAggregator {
+  kind: AggregatorKind.DISTRIBUTION;
+
+  /** Returns snapshot of the current point (value with timestamp). */
+  toPoint(): Point<Distribution>;
+}
+
+/**
+ * HistogramAggregatorType aggregate values into a {@link Histogram} point
+ * type.
+ */
+export interface HistogramAggregatorType extends BaseAggregator {
+  kind: AggregatorKind.HISTOGRAM;
 
   /** Returns snapshot of the current point (value with timestamp). */
-  toPoint(): Point;
+  toPoint(): Point<Histogram>;
 }
 
-export interface Point {
-  value: Sum | LastValue | Distribution | Histogram;
+export type Aggregator =
+  | SumAggregatorType
+  | LastValueAggregatorType
+  | DistributionAggregatorType
+  | HistogramAggregatorType;
+
+/**
+ * Point represents a snapshot of aggregated values of aggregators.
+ */
+export interface Point<T extends PointValueType> {
```

Comment:
why extends, what happens if you don't do it like this 
```
export interface Point<T> {
value: T;
  timestamp: HrTime;	  timestamp: HrTime;
}	}
```

Reply:
This makes the Point accepts any kind of value, which is not expected. The expected value type of Point is the subset of PointValueType, which consists of Sum, LastValue, Distribution, Histogram.